### PR TITLE
disttask: fix panic in task executor/scheduler

### LINKED ISSUE #49869: task executor panic when checking task
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

run task with 2 tidb

as the task has finished and moved into history table
```
[2023/12/28 01:10:02.912 +08:00] [ERROR] [task_executor.go:128] ["BaseTaskExecutor panicked"] [task-id=60003] [recover="runtime error: invalid memory address or nil pointer dereference"] [stack="github.com/pingcap/tidb/pkg/disttask/framework/taskexecutor.(*BaseTaskExecutor).Run.func1
    /home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/pkg/disttask/framework/taskexecutor/task_executor.go:128
runtime.gopanic
    /usr/local/go/src/runtime/panic.go:914
runtime.panicmem
    /usr/local/go/src/runtime/panic.go:261
runtime.sigpanic
    /usr/local/go/src/runtime/signal_unix.go:861
github.com/pingcap/tidb/pkg/disttask/framework/taskexecutor.(*BaseTaskExecutor).run
    /home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/pkg/disttask/framework/taskexecutor/task_executor.go:240
github.com/pingcap/tidb/pkg/disttask/framework/taskexecutor.(*BaseTaskExecutor).Run
    /home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/pkg/disttask/framework/taskexecutor/task_executor.go:136
github.com/pingcap/tidb/pkg/disttask/importinto.(*importExecutor).Run
    /home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/pkg/disttask/importinto/task_executor.go:482
github.com/pingcap/tidb/pkg/disttask/framework/taskexecutor.(*Manager).onRunnableTask
    /home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/pkg/disttask/framework/taskexecutor/manager.go:405
github.com/pingcap/tidb/pkg/disttask/framework/taskexecutor.(*Manager).onRunnableTasks.func1
    /home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/pkg/disttask/framework/taskexecutor/manager.go:230
github.com/pingcap/tidb/pkg/resourcemanager/pool/spool.(*Pool).run.func1
    /home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/pkg/resourcemanager/pool/spool/spool.go:144"]

```

### PR DESCRIPTION
**Pull request by ywqzzy**
### What problem does this PR solve?

Issue Number: close #49869

Problem Summary:

### What changed and how does it work?
Add ErrTaskNotFound when getting tasks.
### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: pkg/disttask/framework/handle/handle.go [hunk 1]
```diff
@@ -52,7 +52,7 @@ func SubmitTask(ctx context.Context, taskKey string, taskType proto.TaskType, co
 		return nil, err
 	}
 	task, err := taskManager.GetTaskByKey(ctx, taskKey)
-	if err != nil {
+	if err != nil || err != storage.ErrTaskNotFound {
```
[the remark this answers is not in the document]
Comment (D3Hunter) [reply pkg/disttask/framework/handle/handle.go:55]: we should return err if task already exists

**Review by D3Hunter — COMMENTED** [review]
rest lgtm

**Review by D3Hunter — APPROVED** [review]

**Review by okJiang — APPROVED** [review]

**Pull request merged**